### PR TITLE
split into multiple module project

### PR DESCRIPTION
**Pull request by XenoAmess**

### CODE REVIEW

**Comment by XenoAmess** [comment]
only split by module without other changes.
@slawekjaranowski

File: version-maven-plugin-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  <inceptionYear>2008</inceptionYear>
+  <licenses>
+    <license>
+      <name>Apache License, Version 2.0</name>
+      <url>https://www.apache.org/licenses/LICENSE-2.0.txt</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+
+  <developers>
```
Comment (slawekjaranowski) [inline version-maven-plugin-api/pom.xml:27]: only in parent

File: version-maven-plugin-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    <groupId>org.codehaus.mojo</groupId>
+    <artifactId>versions-maven-plugin-parent</artifactId>
+    <version>2.11.1-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>versions-maven-plugin-api</artifactId>
+
+  <name>Versions Maven Plugin Api</name>
+  <url>http://www.mojohaus.org/versions-maven-plugin/</url>
```
[the remark this answers is not in the document]
Comment (XenoAmess) [reply version-maven-plugin-api/pom.xml]: @slawekjaranowski done.

File: version-maven-plugin-api/pom.xml (hunk 1 above)
Comment (XenoAmess) [reply version-maven-plugin-api/pom.xml]: @slawekjaranowski done.

File: version-maven-plugin-api/pom.xml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    Versions Plugin for Maven. The Versions Plugin updates the versions of components in the POM.
+  </description>
+  <inceptionYear>2008</inceptionYear>
+  <licenses>
+    <license>
+      <name>Apache License, Version 2.0</name>
+      <url>https://www.apache.org/licenses/LICENSE-2.0.txt</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
```
[the remark this answers is not in the document]
Comment (XenoAmess) [reply version-maven-plugin-api/pom.xml]: @slawekjaranowski done.

**Comment by XenoAmess** [comment]
@slawekjaranowski done.
please review again.

**Comment by slawekjaranowski** [comment]
we should address issues:
- we can not change path for generated xsd files
- api - should contains only interfaces, enums , ...
- api - should have minimal dependencies - even no additional dependencies
- we need check project site after such change

**Comment by slawekjaranowski** [comment]
https://groups.google.com/g/mojohaus-dev/c/hRltMfTmB4o/m/CuRRV7fdAAAJ

**Comment by slawekjaranowski** [comment]
Done in #758 @XenoAmess thanks for your effort.

**Pull request closed without merging**